platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/WritableRosterStore.java

```diff
@@ -0,0 +1,203 @@
+        final Bytes activeRosterHash = RosterUtils.hash(roster).getBytes();
+        if (!roundRosterPairs.isEmpty()) {
+            final RoundRosterPair activeRosterPair = roundRosterPairs.getFirst();
+            if (round < 0 || round <= activeRosterPair.roundNumber()) {
+                throw new IllegalArgumentException(
+                        "incoming round number must be greater than the round number of the current active roster.");
+            }
+        }
+        roundRosterPairs.addFirst(new RoundRosterPair(round, activeRosterHash));
+
+        if (roundRosterPairs.size() > MAXIMUM_ROSTER_HISTORY_SIZE) {
+            final RoundRosterPair lastRemovedRoster = roundRosterPairs.removeLast();
+            removeRoster(lastRemovedRoster.activeRosterHash());
+
+            // At this phase of the implementation, the roster state has a fixed size limit for active rosters.
+            // Future implementations (e.g. DAB) can modify this.
+            if (roundRosterPairs.size() > MAXIMUM_ROSTER_HISTORY_SIZE) {
+                // additional safety check to ensure that the roster state does not contain more than set limit.
+                throw new IllegalStateException(
+                        "Active rosters in the Roster state cannot be more than  " + MAXIMUM_ROSTER_HISTORY_SIZE);
+            }
+        }
+
+        final Builder newRosterState = previousRosterState
+                .copyBuilder()
+                .candidateRosterHash(Bytes.EMPTY)
+                .roundRosterPairs(roundRosterPairs);
+        // since a new active roster is being set, the existing candidate roster is no longer valid
+        // so we remove it if it meets removal criteria.
+        removeRoster(previousRosterState.candidateRosterHash());
+        storeRoster(roster, activeRosterHash, newRosterState);
+    }
+
+    /**
+     * Returns the roster state or throws an exception if the state is null.
+     * @return the roster state
+     * @throws NullPointerException if the roster state is null
+     */
+    @NonNull
+    private RosterState rosterStateOrThrow() {
+        return requireNonNull(rosterState.get());
+    }
+
+    /**
+     * Stores the roster in the roster map, updates the roster store with the provided builder, and commits the changes.
+     *
+     * @param rosterToStore              the roster to store
+     * @param rosterHash                   the hash of the roster
+     * @param rosterStateBuilder       the roster state builder
+     */
+    private void storeRoster(
+            @NonNull final Roster rosterToStore,
+            @NonNull final Bytes rosterHash,
+            @NonNull final Builder rosterStateBuilder) {
+
+        this.rosterState.put(rosterStateBuilder.build());
+        this.rosterMap.put(ProtoBytes.newBuilder().value(rosterHash).build(), rosterToStore);
+
+        if (writableStates instanceof final CommittableWritableStates committableWritableStates) {
+            committableWritableStates.commit();
```

Comment:
The store **must not** commit. The business logic must handle this because otherwise, we cannot roll back changes. In addition, it will prevent other critical optimizations that we plan to implement.

Reply:
removed the commits.